src/view/components/extensionView.jsx

```diff
@@ -93,15 +93,19 @@ const ExtensionView = ({
     init({ initInfo: _initInfo }) {
       setInitInfo(_initInfo);
     },
-    getSettings() {
+    async getSettings() {
       if (!viewRegistrationRef.current) {
         return {};
       }
-
-      return viewRegistrationRef.current.getSettings({
-        initInfo,
-        values: formikPropsRef.current.values
-      });
+      try {
+        return await viewRegistrationRef.current.getSettings({
+          initInfo,
+          values: formikPropsRef.current.values
+        });
+      } catch (e) {
+        reportAsyncError(e);
+        return {};
```

Comment:
I guess I'm not entirely sure what the return value should be here. If we return an empty object, I believe Launch will try to validate it against the schema in our extension.json. If we have any required fields, that would prevent the empty object from actually being saved. But, if we don't have any required fields, then I think an empty object would be saved, which wouldn't be good because it wouldn't represent the actual form values. Re-throwing the error might actually be best.